Timeout in unit test `tests.cmdline.commands.test_processes.py:test_pause_play_kill` when running locally

The test will timeout:
```
(aiida_dev) sph@bastion:~/code/aiida/env/dev/aiida-core$ pytest tests/cmdline/commands/test_process.py  -sv -k test_pause_play_kill
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.7.6, pytest-5.4.1, py-1.5.4, pluggy-0.13.1 -- /home/sph/.virtualenvs/aiida_dev/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/sph/code/aiida/env/dev/aiida-core, inifile: pytest.ini
plugins: regressions-1.0.6, timeout-1.3.4, asyncio-0.10.0, datadir-1.3.1, cov-2.8.1
collected 10 items / 9 deselected / 1 selected                                                                                                                                                                  

tests/cmdline/commands/test_process.py::TestVerdiProcessDaemon::test_pause_play_kill ['/home/sph/.virtualenvs/aiida_dev/bin/verdi', '-p', 'test_profile', 'devel', 'run_daemon']
FAILED

=================================================================================================== FAILURES ====================================================================================================
__________________________________________________________________________________ TestVerdiProcessDaemon.test_pause_play_kill __________________________________________________________________________________

self = <tests.cmdline.commands.test_process.TestVerdiProcessDaemon testMethod=test_pause_play_kill>

    def test_pause_play_kill(self):
        """
        Test the pause/play/kill commands
        """
        # pylint: disable=no-member
        from aiida.orm import load_node
    
        calc = self.runner.submit(test_processes.WaitProcess)
        start_time = time.time()
        while calc.process_state is not plumpy.ProcessState.WAITING:
            if time.time() - start_time >= self.TEST_TIMEOUT:
>               self.fail('Timed out waiting for process to enter waiting state')
E               AssertionError: Timed out waiting for process to enter waiting state
```

Note that this happens both when specifying an explicit test profile as well as when letting the test manager generate one on the fly.

--- a/tests/cmdline/commands/test_process.py
+++ b/tests/cmdline/commands/test_process.py
@@ -8,7 +8,6 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Tests for `verdi process`."""
-
 import datetime
 import subprocess
 import sys
@@ -41,13 +40,19 @@ class TestVerdiProcessDaemon(AiidaTestCase):
 
     def setUp(self):
         super().setUp()
-        from aiida.manage.configuration import get_config
+        from aiida.cmdline.utils.common import get_env_with_venv_bin
         from aiida.engine.daemon.client import DaemonClient
+        from aiida.manage.configuration import get_config
+
+        # Add the current python path to the environment that will be used for the daemon sub process. This is necessary
+        # to guarantee the daemon can also import all the classes that are defined in this `tests` module.
+        env = get_env_with_venv_bin()
+        env['PYTHONPATH'] = ':'.join(sys.path)
 
         profile = get_config().current_profile
         self.daemon_client = DaemonClient(profile)
         self.daemon_pid = subprocess.Popen(
-            self.daemon_client.cmd_string.split(), stderr=sys.stderr, stdout=sys.stdout
+            self.daemon_client.cmd_string.split(), stderr=sys.stderr, stdout=sys.stdout, env=env
         ).pid
         self.runner = get_manager().create_runner(rmq_submit=True)
         self.cli_runner = CliRunner()

